src/vs/platform/terminal/common/capabilities/commandDetection/promptInputModel.ts

```diff
@@ -367,10 +371,10 @@ export class PromptInputModel extends Disposable implements IPromptInputModel {
 			line = buffer.getLine(y);
 			const lineText = line?.translateToString(true);
 			if (lineText && line) {
-				if (this._continuationPrompt === undefined || this._lineContainsContinuationPrompt(lineText)) {
+				if (this._continuationPrompt === undefined && this._shellType !== PosixShellType.Fish || this._lineContainsContinuationPrompt(lineText)) {
```

Comment:
Inverting to === Fish would be better imo, even if we need to duplicate things a little. If we do this consistently it'll be easier to parse the code.